### ISSUE TITLE
Server crash when breaking a pipe with pending orders

### ISSUE THREAD

**Issue by hErbsDK** [RS485/LogisticsPipes#981]
I have had some issues with crafting pipes being placed in wrong spots on some machines, and then the crafting never actually taking place. However the crafting order is still pending. Since there's no cancel feature _(a previous issue discussed this #870 )_ I think to myself: "Whelp! I'll just break the pipe and reset the system" ... and then crash happens :) .. I don't have any crash report right now, but I should be able to recreate if needed.

_This is in FTB InfinityEvolvedSkyblock 1.10 server with nothing changed (the new black :P) - I understand I can just reset the server, but would rather not have to ;) _

**Comment by SonOfTheStars** [comment]
Not much we can do without either a Crash Report OR your complete world and instance for testing purposes...

It would be great if you can provide the Report!

**Comment by hron84** [comment]
Yeah. please provide us a crash report, and also at least a screenshot about your setup. "some machines" is always a tricky expression, a vanilla furnace fells into this category too, but I'm pretty sure your case is not about vanilla funaces. So, please be as verbose as you can.

**Comment by SonOfTheStars** [comment]
@hron84 I dont think the specific machine really Matters. 
Id rather suspect a missing NULL Check or catch. Most likely wihtoin the new enhanced Request Tree that handles extras OR the inventory Side Cache

**Comment by davboecki** [comment]
We still need a crash report to look into this.

**Comment by hErbsDK** [comment]
So I posted this, and then forgot all about it again :) .. It has happened a few times since, but I am not sure I can force it .. Hmmm one sec

<insert some searching time here>

Ok .. I found something I saved ([pastebin](http://pastebin.com/0jdZQTtG)) but not sure it will help .. If it doesn't then I will have to remember next time it happens .. ooo look a squirrel!!!

Ok seriously now, sorry :) .. E.G.: I forget to change satelitte numbers when moving a crafting module, and then send a crafting request .. Nothing happens, and if it's a complex request (i.e. multiple crafting steps) it will halt the whole system of course. Sparkles going off on the crafting pipes waiting, so I try and look to see how bad it is with the Pipe Controller -> click the 5th tab, and get disconnected. I would've broken the pipe to reset things, but I learned from the crash, so I'm reluctant to try that.

I know this changes things a bit, and I'm sorry if I'm not explaining things well enough ...

Hope this helps a bit ...

**Comment by bziemons** [comment]
Thanks for that report. That should already help a bit.

**Comment by hron84** [comment]
@hErbsDK look, a cat: :cat2:  :smile: 

@Bitterholz there is too much... errm.... "machine" in the game, so I prefer to not assume anything. I saw a lot of things where I thought "this should be machine-independent" and really, it wasn't. So I decided to not give a chance for an accidental mistake.

**Comment by hErbsDK** [comment]
@hron84:
they're crazy those cats .. they own the interwebz you know :P 

Also the "machine" is pretty much everything - I have had this happen on chests, LP crafting tables, Railcraft Crusher and well .. more .. (so many machines needed to get through IE:S ...)

The issue is come to think of it mostly related to the crafting pipe waiting for the content it needs from, let's say, the inventory it's connected to. I hope I'm not the only one ending up messing something up with a satelitte, or some machine not in the right setting ... well the list is long, and I've probably been through it a few times .. "This is how we learn!"

also .. ,,,,,,,,,,,,,,,,,,,,, here's some commas to throw in the sentence above

**Comment by hron84** [comment]
@hErbsDK thanks for clarification.
